plugins/generator-1.19.2/forge-1.19.2/templates/livingentity/livingentity.java.ftl

```diff
@@ -391,15 +516,48 @@ public class ${name}Entity extends ${extendsClass} <#if data.ranged>implements R
     }
     </#if>
 
-	<#if hasProcedure(data.onRightClickedOn) || data.ridable || (data.tameable && data.breedable) || (data.guiBoundTo?has_content && data.guiBoundTo != "<NONE>")>
+	<#if hasProcedure(data.onRightClickedOn) || data.ridable || (data.tameable && data.breedable) || (data.guiBoundTo?has_content && data.guiBoundTo != "<NONE>") || data.canTrade>
 	@Override public InteractionResult mobInteract(Player sourceentity, InteractionHand hand) {
 		ItemStack itemstack = sourceentity.getItemInHand(hand);
 		InteractionResult retval = InteractionResult.sidedSuccess(this.level.isClientSide());
 
-		<#if data.guiBoundTo?has_content && data.guiBoundTo != "<NONE>">
+		<#if (data.guiBoundTo?has_content && data.guiBoundTo != "<NONE>") || data.canTrade>
 			<#if data.ridable>
 				if (sourceentity.isSecondaryUseActive()) {
 			</#if>
+			<#if data.canTrade>

```

Comment:
Now change it for a procedure condition 🙃

Reply:
? What do you mean?

Reply:
Instead of using a normal checkbox, users should be able to use a procedire returning a logic value

Reply:
Done